### PR TITLE
 Remove Time.now dependency and duplicate timeout check

### PR DESCRIPTION
**Pull request by indrekj**

### CODE REVIEW

File: spec/freddy/sync_response_container_spec.rb [hunk 1]
```diff
@@ -7,7 +7,15 @@
     subject { container.wait_for_response(0.01) }
 
     it 'raises timeout error' do
-      expect { subject }.to raise_error(Timeout::Error, 'execution expired')
+      begin
```
Comment (willmore) [inline spec/freddy/sync_response_container_spec.rb]: Why not use the previous `expect ... raise_error`?

Comment (indrekj) [inline spec/freddy/sync_response_container_spec.rb]: I also want to check the exception `#response`. With `raise_error` I can only check the exception class and exception `#message`.

Comment (willmore) [inline spec/freddy/sync_response_container_spec.rb]: Does this help?
https://www.relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/raise-error-matcher#set-expectations-on-error-object-passed-to-block

Comment (indrekj) [inline spec/freddy/sync_response_container_spec.rb]: It does, fixed

**Comment by willmore** [comment]
+1

File: lib/freddy/request_manager.rb [hunk 2]
```diff
@@ -1,20 +1,10 @@
...
-    def start
-      @timeout_thread = Thread.new do
-        while true do
-          clear_timeouts Time.now
-          sleep 0.05
-        end
-      end
-    end
-
     def no_route(correlation_id)
```
Comment (stenver) [inline lib/freddy/request_manager.rb]: No tests for request manager?

File: lib/freddy/producers/send_and_wait_response_producer.rb [hunk 3]
```diff
@@ -26,20 +25,30 @@ def initialize(channel, logger)
...
 
+        container.on_timeout do
+          @logger.warn "Request timed out waiting response from #{destination}, correlation id #{correlation_id}"
+          Utils.notify 'RequestTimeout', "Request timed out waiting for response from #{destination}", {
+            correlation_id: correlation_id,
+            destination: destination,
+            timeout_in_seconds: timeout_in_seconds
+          }
+          @request_manager.delete(correlation_id)
+        end
```
Comment (stenver) [inline lib/freddy/producers/send_and_wait_response_producer.rb]: No test to test if request manager deletes the request?

File: lib/freddy/sync_response_container.rb [hunk 4]
```diff
@@ -20,22 +24,18 @@ def wait_for_response(timeout)
       end
 
       if !defined?(@response)
-        raise Timeout::Error, 'execution expired'
+        @on_timeout.call if @on_timeout
```
Comment (stenver) [inline lib/freddy/sync_response_container.rb]: On what condition can this be false?

File: lib/freddy/request_manager.rb (hunk 2 above)
Comment (indrekj) [inline lib/freddy/request_manager.rb]: It has integration tests through freddy_spec.rb

File: lib/freddy/sync_response_container.rb (hunk 4 above)
Comment (indrekj) [inline lib/freddy/sync_response_container.rb]: Currently only in tests

File: lib/freddy/producers/send_and_wait_response_producer.rb (hunk 3 above)
Comment (indrekj) [inline lib/freddy/producers/send_and_wait_response_producer.rb]: there's no flow for request manager to delete request with also invoking timeout

File: lib/freddy/sync_response_container.rb (hunk 4 above)
Comment (stenver) [inline lib/freddy/sync_response_container.rb]: Couldnt we simply mock the timeout call to be nothing so we can remove this confusing line?
```
subject.on_timeout { }
```

Comment (indrekj) [inline lib/freddy/sync_response_container.rb]: updated

**Comment by stenver** [comment]
!squash

**Pull request merged**